base/cvd/BUILD.libwebrtc.bazel

```diff
@@ -0,0 +1,3310 @@
+package(
+    default_visibility = [
+        "@//:android_cuttlefish",
+    ],
+)
+
+cc_library(
+    name = "libwebrtc",
+    srcs = [
+        "api/adaptation/resource.cc",
```

Comment:
Was this file automatically generated in some way? The combination of huge file lists and `select` blocks seems difficult to maintain.

Reply:
Yes, but not in a way that can be reproduced later. I basically copied the Android.bp file and modified it to become a python script that generated this file. The Android.bp file doesn't include headers, so it was easier to merge hundreds (thousands?) of targets into one and add headers to it than to try to find which target each header goes to.

That Android.bp file is generated from a script itself, I plan to use that script to generate build files for more recent versions of libwebrtc (to more closely follow the internal version which is basically the main branch), but I chose to start with the same version as in aosp because that's the one we know already works.

Reply:
If you're interested in running that script at build time, I found it was possible to write `Android.bp` files at external repository loading time [using starlark rules](https://github.com/google/android-cuttlefish/blob/793292a027951c7d02cc4bd44b20fd5dca44dac7/base/cvd/toolchain/file_detector.bzl) but I'm not sure if it's possible to invoke `python`. Maybe with [`repository_ctx.execute`](https://bazel.build/rules/lib/builtins/repository_ctx#execute).

Reply:
Parsing .gn files is not easy. The semi-automated process I know of actually builds with `gn` but generates a json file describing all the targets actually being built for the given configuration (arch, os, etc), then parsed those json files to generate the corresponding build file. Doing something like that at external repository loading time would be possible but inconvenient for webrtc because checking out the code requires around 24GB of disk space, that need to be downloaded first.